### PR TITLE
feat(setup): add setup wizard and script runner

### PR DESCRIPTION
**Pull request by jezekra1**
Add setup script for bee-stack:

Ref: #6 #15 #5

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -74,7 +74,7 @@ web_modules/
 .yarn-integrity
 
 # dotenv environment variable files
-.env
+.env*
```
Comment (Tomas2D) [inline .gitignore:77]: This would ignore all examples.

**Review by Tomas2D — COMMENTED** [review]
I like it. However, I would prefer the JS version for better usability and cross compatibility.

File: .gitignore (hunk 1 above)
Comment (jezekra1) [reply .gitignore:77]: example is named `example.env` -> it should not be a hidden file

Comment (jezekra1) [reply .gitignore:77]: > I like it. However, I would prefer the JS version for better usability and cross compatibility.

This is not feasible, people have trouble installing and configuring docker, `node` / `nvm` / `npm` is an unreasonable requirement for stack and it's not pre-installed on any OS

**Review by planetf1 — CHANGES_REQUESTED** [review]
Validation fails for me with podman desktop (both with the podman compose provider, and docker compose provider).

Details [here](https://github.com/i-am-bee/bee-stack/issues/6#issuecomment-2461818330)

File: .gitignore (hunk 1 above)
Comment (Tomas2D) [reply .gitignore:77]: I know. Therefore we should test this on other platforms like Linux / Windows.

**Comment by jezekra1** [comment]
@planetf1 Do you have docker or rancher installed as well? It's preferred in the script. Should I add another question to select a docker provider? I think most people have only one

**Comment by planetf1** [comment]
> @planetf1 Do you have docker or rancher installed as well? It's preferred in the script. Should I add another question to select a docker provider? I think most people have only one

I don't have docker (desktop) installed on macOS due to licensing/policy + technically too easy to get into messy clashes, especially when using podman in compatibility mode. 

Similarly for rancher, though that remains installed (without compatibility set up)

I do have it on linux

podman has been a bit tricky in the past, but the most recent releases are a lot better with compatibility - most things now work. It's also the most license-friendly implementation (rancher is also good), so I think it's quite important to have support for .

podman also supports either the docker-compose provider, or it's internal one. Ideally it would be better to support either (especially as the plugin then becomes a global configuration). Neither seemed to past the test when I tried the PR earlier :-(

I can take a look at the script checks in more detail if it helps.

**Comment by jezekra1** [comment]
We cannot support outdated (v1) compose commands, the `docker-compose` with dash has reached end of life:
https://docs.docker.com/compose/install/

We support these, the appropriate should now be selected automatically if the version meets the requirements (>= 2.29):
```
podman compose
docker compose
```

You are of course free to use the manual section and run it however you wish, it's a compose file after all...
But the script will not support it

**Review by Tomas2D — APPROVED** [review]

**Pull request merged**